django/contrib/admin/options.py

```diff
@@ -1944,7 +1944,24 @@ def _create_formsets(self, request, obj, change):
                     'files': request.FILES,
                     'save_as_new': '_saveasnew' in request.POST
                 })
-            formsets.append(FormSet(**formset_params))
+            formset = FormSet(**formset_params)
+
+            def _should_delete(request, obj, formset, index):
+                """Return whether or not the user deleted the form."""
+                return (
+                    inline.has_delete_permission(request, obj) and
+                    '{}-{}-DELETE'.format(formset.prefix, index) in request.POST
+                )
+
+            # A "view-only" inline does not submit data. Unless the inline was
```

Comment:
`# Bypass validation of each view-only inline form (since the form's data won't be in request.POST), unless the form was deleted.`
(coding style guideline: write comments without 'we')